keras_nlp/layers/bert_packer.py

```diff
@@ -0,0 +1,228 @@
+                "supported. Received %s" % truncator
+            )
+        self.truncator = truncator
+        self.start_value = start_value
+        self.end_value = end_value
+        self.pad_value = pad_value
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "sequence_length": self.sequence_length,
+                "start_value": self.start_value,
+                "end_value": self.end_value,
+                "pad_value": self.pad_value,
+                "truncator": self.truncator,
+            }
+        )
+        return config
+
+    def _sanitize_inputs(self, inputs):
+        """Force inputs to a list of rank 2 ragged tensors."""
+        # Sanitize inputs.
+        if not isinstance(inputs, (list, tuple)):
+            inputs = [inputs]
+        if not inputs:
+            raise ValueError("At least one input is required for packing")
+        input_ranks = [x.shape.rank for x in inputs]
+        if not all(0 < rank < 3 for rank in input_ranks):
+            raise ValueError(
+                "All inputs for packing must have rank 1 or 2. "
+                f"Received input ranks: {input_ranks}"
+            )
+        if None in input_ranks or len(set(input_ranks)) > 1:
+            raise ValueError(
+                "All inputs for packing must have the same rank. "
+                f"Received input ranks: {input_ranks}"
+            )
+        return inputs
+
+    def _trim_inputs(self, inputs):
+        """Trim inputs to desired length."""
+        num_special_tokens = len(inputs) + 1
+        if self.truncator == "round_robin":
+            return tf_text.RoundRobinTrimmer(
+                self.sequence_length - num_special_tokens
+            ).trim(inputs)
+        elif self.truncator == "waterfall":
+            return tf_text.WaterfallTrimmer(
+                self.sequence_length - num_special_tokens
+            ).trim(inputs)
+        else:
+            raise ValueError("Unsupported truncator: %s" % self.truncator)
+
+    def _combine_inputs(self, segments):
+        """Combine inputs with start and end values added."""
+        dtype = segments[0].dtype
+        start_value = tf.convert_to_tensor(self.start_value, dtype=dtype)
+        end_value = tf.convert_to_tensor(self.end_value, dtype=dtype)
+
+        start_column = tf.tile([start_value], [segments[0].nrows()])
+        start_column = tf.expand_dims(start_column, 1)
+        end_column = tf.tile([end_value], [segments[0].nrows()])
+        end_column = tf.expand_dims(end_column, 1)
+        ones_column = tf.ones_like(start_column, dtype=tf.int32)
+
+        segments_to_combine = [start_column]
+        segment_ids_to_combine = [ones_column * 0]
+        for i, seg in enumerate(segments):
+            # Combine all segments adding end tokens.
+            segments_to_combine.append(seg)
+            segments_to_combine.append(end_column)
```

Comment:
I would prefer renaming `end_token` and `end_column` to `sep_token` and `sep_column`, since this is marking the end of the previous sentence and start of the next one. WDYT?

Reply:
I think we should be consistent across the layer. If we call it "end" in the argument name, no sense renaming here. It still does mark the end of a sentence (only sometime the beginning of a new one).

I think the options are cls and sep, or start and end. Preferred the latter as they felt more readable.